### PR TITLE
APP-3031 Activity API Documentation

### PR DESCRIPTION
**Pull request by symphony-thibault**
Reference documentation for the new Activity API. Added to file [`./doc/activity.md`](https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/blob/2b7fdba88c27782485a032a7c3e8500c5b29cdbd/docs/activity.md)

### CODE REVIEW

File: docs/activity.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-At the moment, two different kinds of activities are supported by the BDK:
+The Activity API is an abstraction built on top of the Datafeed's [_Real Time Events_](https://developers.symphony.com/restapi/docs/real-time-events). An Activity is basically a user interaction triggered from the chat.
+Two different kinds of activities are supported by the BDK:
 - **Command Activity**: triggered when a message is sent in an `IM`, `MIM` or `Chatroom`
 - **Form Activity**: triggered when a user replies to an [_Elements_](https://developers.symphony.com/symphony-developer/docs/overview-of-symphony-elements) form message
 
-Using the Activity API will help you to make your bot interactions easier and faster to implement.
+## Activity Registry
+The main component for activities is certainly the [`ActivityRegistry`](../symphony-bdk-core/src/main/java/com/symphony/bdk/core/activity/ActivityRegistry.java).
+This central component is used to either adding or retrieving activities.
```
Comment (symphony-youri) [inline docs/activity.md:10]: to either add or retrieve? (sounds better to me, not 100% sure)

**Review by symphony-youri — APPROVED** [review]

File: docs/activity.md [hunk 2]
```diff
@@ -1,51 +1,172 @@
...
+    }));
+
+    // finally, start the datafeed loop
+    bdk.datafeed().start();
   }
 }
 ```
+1. `/hello` is the command name 
+2. `true` means that the bot has to be mentioned
+3. the command callback provides the
```
Comment (el-youness) [inline docs/activity.md:106]: Probably forgot to finish the sentence.

File: docs/activity.md [hunk 3]
```diff
@@ -1,51 +1,172 @@
...
-    // TODO
+  @Override
+  protected ActivityInfo info() {
+    final ActivityInfo info = ActivityInfo.of(ActivityType.FORM);
+    info.setName("Hello Form Reply Activity");
+    return info;
   }
 }
 ```
+1. The `ActivityMatcher` here ensure that activity logic is triggered only when the form "**hello-form**" has been
```
Comment (el-youness) [inline docs/activity.md:170]: The `ActivityMatcher` **ensures** that

File: docs/activity.md [hunk 4]
```diff
@@ -1,51 +1,172 @@
...
+  @Override
+  protected ActivityInfo info() {
+    final ActivityInfo info = ActivityInfo.of(ActivityType.FORM);
+    info.setName("Hello Form Reply Activity");
+    return info;
   }
 }
 ```
+1. The `ActivityMatcher` here ensure that activity logic is triggered only when the form "**hello-form**" has been 
+submitted from action button "**submit**"
```
Comment (el-youness) [inline docs/activity.md:171]: from **the** action button "submit"

**Pull request merged**